### PR TITLE
Create a webpack skeleton for React APP

### PR DESCRIPTION
**Pull request by MozdolevskyiMaksym**
Webpack Skeleton for our React App is ready.
Also, I have added additional language extras

### CODE REVIEW

**Review by Winchestery — APPROVED** [review]

**Review by PavelKundenko — APPROVED** [review]

**Review by Icemandov — APPROVED** [review]

**Review by so2niko — APPROVED** [review]

**Review by Anastasiia0999 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Default ignored files
-/shelf/
-/workspace.xml
-/.idea
+shelf/
+workspace.xml
```
Comment (sarhan-azizov) [inline .gitignore:3]:
```suggestion
shelf/
workspace.xml
```
Could u please explain why did you add the resources above?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  mode: "development",
```
Comment (sarhan-azizov) [inline webpack.config.js:5]:
```suggestion
  mode: process.env.NODE_ENV,    
```
Add please create env folder and  .env.development &  .env.production files which should includes env variables

P.S. use dotenv npm package

```
const env = require('dotenv');

env.config({
  path: `${__dirname}/env/.env.${process.env.NODE_ENV ? process.env.NODE_ENV : 'development'}`,
});```

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (MozdolevskyiMaksym) [reply .gitignore:3]: These resources were in .gitignore by default on branch "master" 
default ignored files

File: .babelrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "presets": ["@babel/preset-env", "@babel/preset-react"],
+  "plugins": [
+    "@babel/plugin-proposal-class-properties", 
+    "@babel/plugin-proposal-object-rest-spread",
+    "@babel/plugin-transform-runtime",
+    "@babel/plugin-proposal-private-property-in-object",
```
[the remark this answers is not in the document]
Comment (MozdolevskyiMaksym) [reply .babelrc:7]: Yes, you are right, we do not need these two plugins.

File: .gitignore (hunk 1 above)
Comment (sarhan-azizov) [reply .gitignore:3]: I see thx

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 2 above)
Comment (sarhan-azizov) [reply webpack.config.js:5]: **out of scope!**

it should be implemented in other PR

**Review by sarhan-azizov — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 2 above)
Comment (MozdolevskyiMaksym) [reply webpack.config.js:5]: Okay, understand

File: .babelrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (MozdolevskyiMaksym) [reply .babelrc]: It is done

**Review by sarhan-azizov — APPROVED** [review]
